src/plugins/ScryptEncrypter.ts

```diff
@@ -68,46 +60,42 @@ export const ScryptEncrypter = {
   async encryptKey({
     key,
     password,
+    salt,
+    nonce,
   }: {
     key: Key;
     password: string;
+    salt?: string;
+    nonce?: Uint8Array;
   }): Promise<EncryptedKey> {
-    const salt = generateSalt();
-
-    if (isLedgerKey(key)) {
-      return Promise.resolve({
-        ...key,
-        encrypterName: NAME,
-        salt,
-      } as EncryptedLedgerKey);
-    }
+    const scryptSalt = salt || generateSalt();
 
-    const { privateKey, ...secretlessKey } = key as PlaintextKey;
+    const { privateKey, ...secretlessKey } = key;
 
-    const nonceBytes = nacl.randomBytes(NONCE_BYTES);
-    const scryptedPass = await scryptPass({ password, salt });
+    const scryptNonce = nonce || nacl.randomBytes(NONCE_BYTES);
```

Comment:
I'd call this `secretboxNonce` or something similar. It's not used in the `scrypt` step, just the `secretbox` step.